### DISCUSSION TITLE
Weights and Biases logger improvements.

### DISCUSSION THREAD

**Discussion by parambharat** [jinfagang/yolov7#194] [Ideas]
Hi @jinfagang : I've been contributing to the wandb logger in this repo. I would like to add a few more features.

Currently wandb supports experiment tracking via wandb.run, hyperpameter tuning via wandb.sweeps, visualisation via (wandb.Image and wandb.Table, model and dataset versioning via wandb.Artifact. 

So far I've been able to add experiment tracking and a few visualisation components to the repository. I would like to add more features for checkpointing and model versioning into the repository. However, this will require some effort and refactoring of the current training and validation loops via a custom trainer and hooks in the detectron framework.

Before i begin work on these features, i would like to confirm if you would be willing to continue accepting changes to the repo that might change the way the trainers are organised. A little detail on what your next focus is might really help me making the decision to dedicate more time to this.

**Comment by lucasjinreal** [comment]
@parambharat Sounds like this involves many features. I am not a fan of wandb, but have better than none, you can try adding features one by one as different PR.
